src/test/java/it/gov/pagopa/pu/debtpositions/service/create/receipt/InstallmentUpdateServiceTest.java

```diff
@@ -34,66 +35,88 @@ class InstallmentUpdateServiceTest {
+        InstallmentStatus.DRAFT, InstallmentStatus.PAID)
     )));
 
     //align entities id
     debtPosition.getPaymentOptions().forEach(paymentOption -> {
       paymentOption.setDebtPositionId(debtPosition.getDebtPositionId());
-      paymentOption.getInstallments().forEach(anInstallment -> anInstallment.setPaymentOptionId(paymentOption.getPaymentOptionId()));
+      paymentOption.getInstallments().forEach(
+        anInstallment -> anInstallment.setPaymentOptionId(
+          paymentOption.getPaymentOptionId()));
     });
 
-    Mockito.when(debtPositionRepositoryMock.findByInstallmentId(targetInstallment.getInstallmentId())).thenReturn(debtPosition);
+    Mockito.when(debtPositionRepositoryMock.findByInstallmentId(
+      targetInstallment.getInstallmentId())).thenReturn(debtPosition);
 
     //when
-    DebtPosition response = installmentUpdateService.updateInstallmentStatusOfDebtPosition(targetInstallment, receiptDTO);
+    DebtPosition response = installmentUpdateService.updateInstallmentStatusOfDebtPosition(
+      targetInstallment, receiptDTO);
 
     //verify
     Assertions.assertEquals(debtPosition, response);
     //find target installment in response
-    InstallmentNoPII responseTargetInstallment = response.getPaymentOptions().stream()
+    InstallmentNoPII responseTargetInstallment = response.getPaymentOptions()
+      .stream()
       .flatMap(paymentOption -> paymentOption.getInstallments().stream())
-      .filter(anInstallment -> anInstallment.getInstallmentId().equals(targetInstallment.getInstallmentId()))
+      .filter(anInstallment -> anInstallment.getInstallmentId()
+        .equals(targetInstallment.getInstallmentId()))
       .findFirst().orElse(null);
     Assertions.assertNotNull(responseTargetInstallment);
     //verify that the target installment has been updated to (TO_SYNC->)PAID
-    verifyInstallmentStatus(responseTargetInstallment, InstallmentStatus.PAID, "target installment status");
+    verifyInstallmentStatus(responseTargetInstallment, InstallmentStatus.PAID,
+      "target installment status");
     //verify that there no more NOT-PAID installments on payment options different of the one of the target installment
     int[] idxPo = {0};
     int[] idxInst = {0};
     debtPosition.getPaymentOptions().forEach(paymentOption -> {
-      if (!paymentOption.getPaymentOptionId().equals(targetInstallment.getPaymentOptionId())) {
+      if (!paymentOption.getPaymentOptionId()
+        .equals(targetInstallment.getPaymentOptionId())) {
         paymentOption.getInstallments().forEach(anInstallment -> {
-          if (InstallmentUpdateService.NOT_PAID.contains(anInstallment.getStatus())) {
-            verifyInstallmentStatus(anInstallment, InstallmentStatus.INVALID,
+          if (InstallmentUtils.isPayable(anInstallment)) {
+            InstallmentStatus expectedStatus = anInstallment.getStatus();
+            verifyInstallmentStatus(anInstallment, expectedStatus,
```

Comment:
here you are changing the test behavior, according to the previous logic it should be:
```suggestion
            verifyInstallmentStatus(anInstallment, InstallmentStatus.INVALID,
```